src/Model.php

```diff
@@ -613,26 +616,49 @@ public function getFields($filter = null): array
     {
         if ($filter === null) {
             return $this->fields;
-        } elseif (is_string($filter)) {
+        } elseif (!is_array($filter)) {
             $filter = [$filter];
         }
 
-        return array_filter($this->fields, function (Field $field, $name) use ($filter) {
-            // do not return fields outside of "only_fields" scope
-            if ($this->only_fields && !in_array($name, $this->only_fields, true)) {
-                return false;
-            }
-            foreach ($filter as $f) {
-                if (
-                    ($f === 'system' && $field->system)
-                    || ($f === 'not system' && !$field->system)
-                    || ($f === 'editable' && $field->isEditable())
-                    || ($f === 'visible' && $field->isVisible())
-                ) {
+        $filterPresets = [
+            'system' => function (Field $field) {
+                return $this->isOnlyFieldsField($field->short_name) && $field->system;
+            },
+            'not system' => function (Field $field) {
+                return $this->isOnlyFieldsField($field->short_name) && !$field->system;
+            },
+            'editable' => function (Field $field) {
+                return $this->isOnlyFieldsField($field->short_name) && $field->isEditable();
+            },
+            'visible' => function (Field $field) {
+                return $this->isOnlyFieldsField($field->short_name) && $field->isVisible();
+            },
+            'persisting' => function (Field $field) {
```

Comment:
`persist` is better name

Reply:
Yes, I was also thinking about declaring in constants, will do